addons/calendar/data/mail_template_data.xml

```diff
@@ -231,44 +278,202 @@
             <div style='font-size: 12px; text-align: center; font-weight: bold; color: #ffffff; background-color: #875A7B;'>
                 ${event_id.get_interval('month', tz=object.partner_id.tz if not event_id.allday else None)}
             </div>
-            <div style="border-collapse: separate; color: #5F5F5F; text-align: center; font-size: 12px; border-bottom-right-radius: 3px; font-weight: bold; border: 1px solid #875A7B; border-bottom-left-radius: 3px;">
+            <div style="color: #5F5F5F; text-align: center; font-size: 12px; font-weight: bold; border: 1px solid #875A7B;">
                 ${not event_id.allday and event_id.get_interval('time', tz=object.partner_id.tz) or ''}
             </div>
+            % if format_date(object.event_id.start) != format_date(object.event_id.stop):
+                    % if object.event_id.allday:
+                        <div style="border-collapse: separate; color: #5F5F5F; text-align: center; font-size: 12px; border-bottom-right-radius: 3px; font-weight: bold; border: 1px solid #875A7B; border-bottom-left-radius: 3px;">
+                            &#8594; ${object.event_id.get_interval('end_day', tz=object.partner_id.tz) or ''}
+                        </div>
+                    % else
+                        <div style='font-size: 12px; text-align: center; font-weight: bold; color: #ffffff; border-bottom-right-radius: 3px; border-bottom-left-radius: 3px; background-color: #875A7B;'>
+                            &#8594; ${object.event_id.get_interval('end_day', tz=object.partner_id.tz) or ''}
+                            ${object.event_id.get_interval('end_time', tz=object.partner_id.tz) or ''}
+                        </div>
+                    % endif
+                % else
+                    <div style='font-size: 12px; text-align: center; font-weight: bold; color: #ffffff; border-bottom-right-radius: 3px; border-bottom-left-radius: 3px; background-color: #875A7B;'>
+                        %if not object.event_id.allday:
```

Comment:
You can use elif and fix indentation to align it with line 284. it will be easier to inderstand.